packages/ui/packages/react/src/hooks.ts

```diff
@@ -29,6 +29,72 @@ export interface Datamodel {
   logout: () => Promise<void>
 }
 
+// KMS Configuration types and hook
+export interface KMSConfig {
+  keyManagerServiceURL: string
+  keyManagerServiceDID: string
+  location?: string
+  keyring?: string
+  /**
+   * If set to true will allow HTTP connections to the KMS server. Default is false.
+   */
+  allowInsecureHttp?: boolean
+}
+
+/**
+ * Hook for managing KMS configuration state.
+ * Provides shared KMS config state that can be used by both upload and decryption flows.
+ * @param initialConfig - Initial KMS configuration
+ */
+export function useKMSConfig(initialConfig?: KMSConfig): {
+  kmsConfig: KMSConfig | undefined
+  setKmsConfig: (config: KMSConfig | undefined) => void
+  createKMSAdapter: () => Promise<any | null>
+  isConfigured: boolean
```

Comment:
question: when is it not configured? given the env variable fallback. if I pass an initial config but it has no value for the keyManagerService?

Reply:
When you explicitly override any of the configs:
```
useKMSConfig({ 
  keyManagerServiceURL: undefined  // Explicitly overrides the fallback
})
```

> isConfigured: Boolean(kmsConfig?.keyManagerServiceURL && kmsConfig?.keyManagerServiceDID)